File: snippets/csharp/Messages.cs
using Breez.Sdk.Liquid;

public class MessagesSnippets
{
    public void SignMessage(BindingLiquidSdk sdk)
    {
        // ANCHOR: sign-message
        var message = "<message to sign>";
        try
        {
            var signMessageRequest = new SignMessageRequest(message);
            var signMessageResponse = sdk.SignMessage(signMessageRequest);

            // Get the wallet info for your pubkey
            var info = sdk.GetInfo();

            var signature = signMessageResponse?.signature;
            var pubkey = info?.walletInfo?.pubkey;

            Console.WriteLine($"Pubkey: {pubkey}");
            Console.WriteLine($"Signature: {signature}");
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: sign-message
    }

    public void CheckMessage(BindingLiquidSdk sdk)
    {
        // ANCHOR: check-message
        var message = "<message>";
        var pubkey = "<pubkey of signer>";
        var signature = "<message signature>";
        try
        {
            var checkMessageRequest = new CheckMessageRequest(message, pubkey, signature);
            var checkMessageResponse = sdk.CheckMessage(checkMessageRequest);

            var isValid = checkMessageResponse?.isValid;

            Console.WriteLine($"Signature valid: {isValid}");
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: check-message
    }
}

File: snippets/wasm/messages.ts
import { type BindingLiquidSdk } from '@breeztech/breez-sdk-liquid'

const exampleSignMessage = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: sign-message
  const signMessageResponse = sdk.signMessage({
    message: '<message to sign>'
  })

  // Get the wallet info for your pubkey
  const info = await sdk.getInfo()

  const signature = signMessageResponse.signature
  const pubkey = info.walletInfo.pubkey

  console.log(`Pubkey: ${pubkey}`)
  console.log(`Signature: ${signature}`)
  // ANCHOR_END: sign-message
}

const exampleCheckMessage = (sdk: BindingLiquidSdk) => {
  // ANCHOR: check-message
  const checkMessageResponse = sdk.checkMessage({
    message: '<message>',
    pubkey: '<pubkey of signer>',
    signature: '<message signature>'
  })
  const isValid = checkMessageResponse.isValid

  console.log(`Signature valid: ${isValid}`)
  // ANCHOR_END: check-message
}
